File: cpp/Guarana.cpp
#include "Guarana.h"
#include "Komentarz.h"

Guarana::Guarana(Swiat *swiat, Punkt pole, int turaPowstania)
    : Roslina(swiat, pole, turaPowstania, sila, inicjatywa, symbol){
    this->symbol = SYMBOL_GUARANY;
    this->sila = SILA_GUARANY;
    this->inicjatywa = INICJATYWA_GUARANY;
    this->pole = pole;
    this->swiat = swiat;
    this->umarl = false;
    this->turaPowstania = turaPowstania;
}

void Guarana::kolizja(Organizm *innyOrganizm) {
    swiat->zabijOrganizm(this);
    if(this->sila >= innyOrganizm->getSila()) {
        swiat->zabijOrganizm(innyOrganizm);
        Komentarz::dodajKomentarz(innyOrganizm->nazwaOrganizmu() + " zjada " + this->nazwaOrganizmu() + " i ginie");
    }
    else {
        swiat->przemiescOrganizm(innyOrganizm, this->getPole());
        innyOrganizm->setSila(innyOrganizm->getSila() + 3); //specjalna kolizja guarany
        Komentarz::dodajKomentarz(innyOrganizm->nazwaOrganizmu() + " zjada " + this->nazwaOrganizmu() + " i otrzymuje +3 sily");
    }
}

string Guarana::nazwaOrganizmu() {
    return "Guarana";
}

File: java/pl/edu/pg/po/projekt2/rosliny/Guarana.java
package pl.edu.pg.po.projekt2.rosliny;

import pl.edu.pg.po.projekt2.*;

public class Guarana extends Roslina {
    private static final char symbol = 'G';
    private static final int inicjatywa = 0;
    private static final int sila = 0;

    public Guarana(Swiat swiat, Punkt pole, int turaPowstania) {
        super(swiat, pole, turaPowstania, sila, inicjatywa, symbol);
        this.pole = pole;
        this.swiat = swiat;
        this.umarl = false;
        this.turaPowstania = turaPowstania;
    }

    @Override
    public void kolizja(Organizm innyOrganizm) {
        if(sila >= innyOrganizm.getSila()) {
            swiat.zabijOrganizm(this);
            swiat.zabijOrganizm(innyOrganizm);
            Komentarz.dodajKomentarz(innyOrganizm.nazwaOrganizmu() + " zjada " + this.nazwaOrganizmu() + " i ginie");
        }
        else {
            Punkt tmpPole = this.getPole();
            swiat.zabijOrganizm(this);
            swiat.przemiescOrganizm(innyOrganizm, tmpPole);
            innyOrganizm.setSila(innyOrganizm.getSila() + 3); //specjalna kolizja guarany
            Komentarz.dodajKomentarz(innyOrganizm.nazwaOrganizmu() + " zjada " + this.nazwaOrganizmu() + " i otrzymuje +3 sily");
        }
    }

    @Override
    public String nazwaOrganizmu() {
        return "Guarana";
    }
}
